platform/search/src/SearchAggregator.js

```diff
@@ -0,0 +1,144 @@
+                        i -= 1;
+                    } else {
+                        // Otherwise add the ID to the list of the ones we have seen 
+                        ids.push(results[i].id);
+                    }
+                }
+                
+                return numRemoved;
+            }
+            
+            // Order the objects from highest to lowest score in the array.
+            // Modifies the passed array, as well as returns the modified array. 
+            function orderByScore(results) {
+                results.sort(function (a, b) {
+                    if (a.score > b.score) {
+                        return -1;
+                    } else if (b.score > a.score) {
+                        return 1;
+                    } else {
+                        return 0;
+                    }
+                });
+                return results;
+            }
+            
+            // For documentation, see query below.
+            function queryAll(inputText, maxResults) {
+                var i,
+                    timestamp = Date.now(),
+                    resultPromises = [];
+                
+                if (!maxResults) {
+                    maxResults = DEFAULT_MAX_RESULTS;
+                }
+                
+                // Send the query to all the providers
+                for (i = 0; i < providers.length; i += 1) {
+                    resultPromises.push(
+                        providers[i].query(inputText, timestamp, maxResults, DEFUALT_TIMEOUT)
+                    );
+                }
+                
+                // Get promises for results arrays
+                return $q.all(resultPromises).then(function (resultObjects) {
+                    var results = [],
+                        totalSum = 0,
+                        i;
+                    
+                    // Merge results 
+                    for (i = 0; i < resultObjects.length; i += 1) {
+                        results = results.concat(resultObjects[i].hits);
+                        totalSum += resultObjects[i].total;
+                    }
+                    // Order by score first, so that when removing repeats we keep the higher scored ones
+                    orderByScore(results);
+                    totalSum -= filterDuplicates(results, totalSum);
+                    
+                    return {
+                        hits: results,
+                        total: totalSum,
+                        timedOut: resultObjects.some(function (obj) {
+                            return obj.timedOut;
+                        })
+                    };
+                });
+            }
+            
+            return {
+                /** 
+                 * Sends a query to each of the providers. Returns a promise for
+                 *   a result object that has the format
+                 *   {hits: domainObject[], total: number}
```

Comment:
Ah, got it. I think your rationale there is valid, but I'd rather see both of these have the same interface, because (a) it reduces the total number of interfaces, and (b) it leaves the door open for future extensions (such as decorator components) which could, in principle, use those scores.

Put another way, having a consistent interface here would mean that if you went from writing a provider to using the `searchService` (or vice versa) you wouldn't have to learn a subtly different interface to do so.